lib/modules/dosomething/dosomething_northstar/dosomething_northstar.auth.inc

```diff
@@ -51,3 +51,22 @@ function dosomething_northstar_openid_connect_post_authorize($tokens, $account,
 
   user_save($account, $edit);
 }
+
+/**
+ * Redirect to destroy a Northstar single-sign-on session if one exists.
+ *
+ * Implements hook_user_logout().
+ *
+ * @param $account
+ */
+function dosomething_northstar_user_logout($account) {
```

Comment:
One gotcha: because of the `drupal_goto` part of this, this will prevent any other hooks from firing afterwards (and hence why we need to `session_destroy` in here rather that using Drupal's default in `user_logout`).

I don't see a relevant "post-logout" hook but open to suggestions!